CONTRIBUTING.md

```diff
@@ -116,11 +117,42 @@ Below are some conventions that aren't enforced by any of our tooling but we non
    wish to make, if applicable.
 1. Add any appropriate tests.
 1. Make your code or other changes.
-1. Review guidelines such as [How to write the perfect pull request][github-perfect-pr], thanks!
-
-Take a look at [Github Flow][github-flow] for a more detailed explanation of this process.
-
-[angular-commit-style]: https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits
+1. Please adhere to the guidelines in [How to write the perfect pull request][github-perfect-pr], thanks!
+1. Please perform a self-review using the reviewer guidelines below prior to taking the PR out of draft state.
+
+### Reviewer Guidelines
+
+Reviewers should use the following questions to evaluate the implementation for correctness/completeness and ensure all housekeeping items have been addressed prior to merging the code.
+
+- Correctness/completeness
+  1. Do you fully understand the implementation? (Would you be comfortable explaining how this code works to someone else?)
+  1. Does the code meet the acceptance criteria?
+     - If there is an associated spec, does the code match the spec?
+  1. Is the intention of the code captured in relevant tests?
+     - Does the description of each test accurately represent the assertions?
+     - For any test explicitly called out on the ticket as desirable to implement, was it implemented?
+     - If there are prose spec tests, were they implemented?
+     - If there are associated automated spec tests, were they all pulled in and are they all running and correctly interpreting the spec inputs?
+       - Are any runner changes needed to process new input types?
+  1. Could these changes impact any adjacent functionality?
+  1. Are there any errors that might not be correctly caught or propagated?
```

Comment:
Stop me if I'm getting to implementation specific

A sub-bullet here about mocha specifically, testing event handling and how error throwing can make unrelated tests fail.